### ISSUE TITLE
Cart page throws 500 when submitting promo code with invalid session

### ISSUE THREAD

**Issue by ulinskas** [Sylius/Sylius#15151]
**Sylius version affected**: 1.12.8

**Description**  
Sylius cart page throws 500 error when user session is invalid.
Issue appeared after https://github.com/Sylius/Sylius/pull/14865

**Steps to reproduce**  
Tested on Sylius 1.12.8
Also on https://demo.sylius.com/en_US/
1. Add any product to cart
2. Open browser dev tools -> Application -> Cookies 
3. Delete `PHPSESSID` cookie
4. Enter random promo code and hit submit

**Possible Solution**  
`Sylius\Bundle\OrderBundle\Controller::resetChangesOnCart action`
Change from
```
private function resetChangesOnCart(OrderInterface $cart): void
    {
        $this->manager->refresh($cart);
        foreach ($cart->getItems() as $item) {
            $this->manager->refresh($item);
        }
    }
```
To
```
    private function resetChangesOnCart(OrderInterface $cart): void
    {
        if (!$cart->getId()) {
            return;
        }

        $this->manager->refresh($cart);
        foreach ($cart->getItems() as $item) {
            $this->manager->refresh($item);
        }
    }
```

Check if cart Object is new or not before trying to refetch it from DB.

**Comment by jakubtobiasz** [comment]
Hi @ulinskas 👋🏼!
Thanks for reporting. Would you like to open a PR with your suggestion?

**Comment by jakubtobiasz** [comment]
Hi @ulinskas 👋🏼!
Your PR has been merged, and should be released with Sylius `1.12.9` this week :).

**Comment by ulinskas** [comment]
Thank you @jakubtobiasz :)
